### PR TITLE
Jetpack and Powered Armor refactor

### LINKED ISSUE #1232: Keybind to Toggle On/Off Jetpacks
### Cross-mod Integration

_No response_

### Feature Description

Having no way to turn off the jet-pack of the GTM armors is really frustrating if you've have access to another form of flight. The player might not want to fly constantly so an keybind to disable them would be hugely appreciated

### PR DESCRIPTION
**Pull request by YoungOnionMC**
## What
Fixes: #1232 and #1636 

## Implementation Details
lots of refactoring

### CODE REVIEW

File: src/main/java/com/gregtechceu/gtceu/api/block/MaterialBlock.java [hunk 1]
```diff
@@ -279,30 +275,4 @@ public boolean replaceWithFramedPipe(Level level, BlockPos pos, BlockState state
...
-            }
-            if (livingEntity.horizontalCollision) {
-                d2 = 0.3;
-            }
-
-            deltaMovement = new Vec3(d0, d2, d1);
-
-            entity.setDeltaMovement(deltaMovement);
-        }
-    }
```
Comment (screret) [inline src/main/java/com/gregtechceu/gtceu/api/block/MaterialBlock.java:307]: why remove?

File: src/main/java/com/gregtechceu/gtceu/common/item/armor/IJetpack.java [hunk 2]
```diff
@@ -64,60 +67,129 @@ default float getFallDamageReduction() {
...
+            tag.putBoolean("enabled", true);
+        if (tag.contains("hover"))
+            tag.putBoolean("hover", true);
+        player.displayClientMessage(Component.translatable("metaarmor.jetpack.emergency_hover_mode"), true);
+        stack.setTag(tag);
+
+        if (!player.level().isClientSide) {
+            player.fallDistance = 0;
+            if (player instanceof ServerPlayer) {
+                ((ServerPlayer) player).connection.aboveGroundTickCount = 0;
```
Comment (screret) [inline src/main/java/com/gregtechceu/gtceu/common/item/armor/IJetpack.java:184]: do this like `player instanceof ServerPlayer serverPlayer` instead

File: src/main/java/com/gregtechceu/gtceu/common/item/armor/QuarkTechSuite.java [hunk 3]
```diff
@@ -47,6 +47,11 @@ public class QuarkTechSuite extends ArmorLogicSuite implements IStepAssist {
 
     protected static final Map<MobEffect, Integer> potionRemovalCost = new IdentityHashMap<>();
     private float charge = 0.0F;
+    private static final int MIN_NIGHTVISION_CHARGE = 4;
+    private static final int NIGHTVISION_DURATION = 20 * 20; // 20 seconds
+    private static final int NIGHT_VISION_RESET = 11 * 20; // 11 seconds is before the flashing
+    private static final byte RUNNING_TIMER = 10; // .5 seconds
+    private static final double LEGGING_ACCEL = 0.085D;
```
Comment (screret) [inline src/main/java/com/gregtechceu/gtceu/common/item/armor/QuarkTechSuite.java]: why are these duplicated? make them constants in ArmorUtils or such

**Review by screret — CHANGES_REQUESTED** [review]

File: src/main/java/com/gregtechceu/gtceu/api/block/MaterialBlock.java (hunk 1 above)
Comment (YoungOnionMC) [reply src/main/java/com/gregtechceu/gtceu/api/block/MaterialBlock.java:307]: the entire logic just doesnt work atm, we need to figure out a better method for making them exactly ladder like

**Review by screret — APPROVED** [review]

**Pull request merged**